src/XR/webXRRenderTargetTextureProvider.ts

```diff
@@ -45,42 +46,62 @@ export abstract class WebXRLayerRenderTargetTextureProvider implements IWebXRRen
-        const renderTargetTexture = new RenderTargetTexture("XR renderTargetTexture", { width, height }, this._scene);
-        const renderTargetWrapper = renderTargetTexture.renderTarget!;
-        (renderTargetWrapper as WebGLRenderTargetWrapper)._framebuffer = framebuffer;
+        const renderTargetTexture = multiview
+            ? new MultiviewRenderTarget(this._scene, { width, height })
+            : new RenderTargetTexture("XR renderTargetTexture", { width, height }, this._scene);
+        const renderTargetWrapper = renderTargetTexture.renderTarget as WebGLRenderTargetWrapper;
+        // Set the framebuffer, make sure it works in all scenarios - emulator, no layers and layers
+        if (framebuffer || !colorTexture) {
+            renderTargetWrapper._framebuffer = framebuffer;
+        }
 
         // Create internal texture
-        const internalTexture = new InternalTexture(engine, InternalTextureSource.Unknown, true);
-        internalTexture.width = width;
-        internalTexture.height = height;
-        if (!!colorTexture) {
-            internalTexture._hardwareTexture = colorTexture;
+        if (colorTexture) {
+            if (multiview) {
+                renderTargetWrapper._colorTextureArray = colorTexture;
+            } else {
+                const internalTexture = new InternalTexture(engine, InternalTextureSource.Unknown, true);
+                internalTexture.width = width;
+                internalTexture.height = height;
+                internalTexture._hardwareTexture = new WebGLHardwareTexture(colorTexture, this._glContext);
+                renderTargetWrapper.setTexture(internalTexture, 0);
+                internalTexture.isReady = true;
+                renderTargetTexture._texture = internalTexture;
+            }
         }
-        renderTargetWrapper.setTexture(internalTexture, 0);
-        renderTargetTexture._texture = internalTexture;
 
-        if (!!depthStencilTexture) {
-            const internalDSTexture = new InternalTexture(engine, InternalTextureSource.DepthStencil, true);
-            internalDSTexture.width = width;
-            internalDSTexture.height = height;
-            internalDSTexture._hardwareTexture = depthStencilTexture;
-            renderTargetWrapper._depthStencilTexture = internalDSTexture;
+        if (depthStencilTexture) {
+            if (multiview) {
+                renderTargetWrapper._depthStencilTextureArray = depthStencilTexture;
+            } else {
+                const internalDSTexture = new InternalTexture(engine, InternalTextureSource.DepthStencil, true);
+                internalDSTexture.width = width;
+                internalDSTexture.height = height;
+                internalDSTexture._hardwareTexture = new WebGLHardwareTexture(depthStencilTexture, this._glContext);
+                internalDSTexture.isReady = true;
+                renderTargetWrapper._depthStencilTexture = internalDSTexture;
+            }

```

Comment:
nit - it  feels like some code here can be spared. I guess the main thing that can change is the initialization of the internal texture?But I will leave this up to you - if you don't think it is needed, keep it this way

Reply:
To be clear, do you mean moving the repeated code into a function for reuse?

Reply:
yes, feels like this code is repeated. I know it's small, and if you feel like it won't bring any improvement you can ignore this.